op-chain-ops/genesis/layer_two.go

```diff
@@ -46,3 +54,74 @@ func BuildL2DeveloperGenesis(config *DeployConfig, l1StartBlock *types.Block) (*
 
 	return db.Genesis(), nil
 }
+
+// BuildL2MainnetGenesis will build an L2 Genesis suitable for a Superchain mainnet that does not
+// require a pre-bedrock migration & supports optional governance token predeploy.
+func BuildL2MainnetGenesis(config *DeployConfig, l1StartBlock *types.Block) (*core.Genesis, error) {
+	genspec, err := NewL2Genesis(config, l1StartBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	db := state.NewMemoryStateDB(genspec)
+
+	storage, err := NewL2StorageConfig(config, l1StartBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	immutable, err := NewL2ImmutableConfig(config, l1StartBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set up the proxies
+	depBytecode, err := bindings.GetDeployedBytecode("Proxy")
+	if err != nil {
+		return nil, err
+	}
+	if len(depBytecode) == 0 {
+		return nil, errors.New("Proxy has empty bytecode")
+	}
+	for i := uint64(0); i <= 2048; i++ {
```

Comment:
This code is duplicated in a few places, perhaps we can reuse this code?

Reply:
The other proxy deployment loops are slightly different since they also embed the handling of migration cases. I'll take another look and see if that can be cleanly separated out, but I figured the migration code might be short lived anyway.

Reply:
Ah got it, I'd like to remove the migration code to reduce diff in these sorts of PRs since it can live in our legacy repo, then hopefully we can settle on 1 implementation of creating the L2 genesis. I believe there is an `op-node` genesis command that will generate the `genesis.json`, that was the UX we came up with for creating a genesis file

Reply:
op-node's genesis command calls the function above this one (BuildL2DeveloperGenesis), which we should probably replace with one that doesn't include the deprecated stuff.